src/main/scala/cromwell/engine/backend/local/SharedFileSystem.scala

```diff
@@ -130,11 +137,30 @@ trait SharedFileSystem {
       WdlFile(WdlFile.appendPathsWithSlashSeparators("/root", pathWithWorkflowName.subpath(1, pathWithWorkflowName.getNameCount).toString))
     }
 
-    // If this call is using Docker, adjust input paths, otherwise return unaltered input paths.
+    def localizeInputToCallDirectory(path: String): Path = {
+      val originalPath = Paths.get(path)
+      val callPath = Try {
+        val callInputsDirectory = LocalBackend.hostCallPath(workflowDescriptor, call.name, callKey.index).resolve("inputs")
+        if (!callInputsDirectory.toFile.exists) callInputsDirectory.toFile.mkdirs
+        val hash = hashPath(originalPath)
+        callInputsDirectory.resolve(s"$hash-${originalPath.getFileName.toString}")
```

Comment:
An "only when necessary" policy would probably work in most real-world scenarios, though it wouldn't be impossible to break this.  If these are all given as workflow inputs:

/path/to/foo.bam
/path/to/foo.bam.index
/different/path/to/foo.bam.index

The index files would get hashes prepended and would then fail to line up with the bam.

I was thinking we might stage workflow inputs into hashed directories and therefore not manipulate the file names at all, but there could be other ways of solving this as well.